pkg/apis/internal/common.go

```diff
@@ -0,0 +1,39 @@
+package internal
```

Comment:
not a fan of this package name **or** the type `CommonMethods` - one of those should say it's about `GS` or `GenericServices`

Reply:
I'd go with a package `pkg/apis/internal/gs`, moving `CommonMethods`, `State` and `StateRetriever` there

Reply:
maybe call `CommonMethods` something like `GenericService` - this reads better for people knowing what generic services are:

```go
type Cluster struct {
  gs.GenericService

  // [...]
}
```